sorc/chgres_cube.fd/CMakeLists.txt

```diff
@@ -28,7 +28,7 @@ endif()
 set(exe_name chgres_cube)
 add_executable(${exe_name} ${fortran_src})
 target_include_directories(
-  ${exe_name} PRIVATE ${ESMF_INC} ${WGRIB2_INCLUDES})
+  ${exe_name} PRIVATE ${WGRIB2_INCLUDES})
```

Comment:
We should be able to express chgres_cube's dependency on WGRIB2 using imported target as we do with ESMF and all other libraries.

Reply:
If someone wants to update FindWGRIB2.cmake, I will be happy to use it.

Reply:
Try to add these 6 lines at the end of FindWGRIB2.cmake :
```
diff --git a/Modules/FindWGRIB2.cmake b/Modules/FindWGRIB2.cmake
index 702c0a4..5f3376b 100644
--- a/Modules/FindWGRIB2.cmake
+++ b/Modules/FindWGRIB2.cmake
@@ -18,3 +18,9 @@ else()
 endif()
 
 mark_as_advanced (WGRIB2_INCLUDES WGRIB2_LIBRARIES)
+
+add_library(wgrib2 UNKNOWN IMPORTED)
+set_target_properties(wgrib2 PROPERTIES
+  IMPORTED_LOCATION "${WGRIB2_LIBRARIES}"
+  INTERFACE_INCLUDE_DIRECTORIES "${WGRIB2_INCLUDES}"
+  INTERFACE_LINK_LIBRARIES "${WGRIB2_LIBRARIES}")
```
and then in ``sorc/chgres_cube.fd/CMakeLists.txt`` remove ``target_include_directories`` and replace ``${WGRIB2_LIBRARIES}`` with ``wgrib2``.

For example:
```
diff --git a/sorc/chgres_cube.fd/CMakeLists.txt b/sorc/chgres_cube.fd/CMakeLists.txt
index 0b888ce4..36509dcb 100644
--- a/sorc/chgres_cube.fd/CMakeLists.txt
+++ b/sorc/chgres_cube.fd/CMakeLists.txt
@@ -27,8 +27,8 @@ endif()
 
 set(exe_name chgres_cube)
 add_executable(${exe_name} ${fortran_src})
-target_include_directories(
-  ${exe_name} PRIVATE ${WGRIB2_INCLUDES})
+#target_include_directories(
+#  ${exe_name} PRIVATE ${WGRIB2_INCLUDES})
 target_link_libraries(
   ${exe_name}
   nemsio
@@ -38,8 +38,8 @@ target_link_libraries(
   sp_d
   w3nco_d
   esmf
+  wgrib2
   MPI::MPI_Fortran
-  ${WGRIB2_LIBRARIES}
   NetCDF::NetCDF_Fortran
   OpenMP::OpenMP_Fortran)
```

Reply:
I tested your change and it works.  I can initiate a pull request for the updated FindWGRIB2.cmake.

Reply:
Updated the branch to point to Dusan's improved FindWGRIB2.cmake.